### PR TITLE
Slider reduce slop when active

### LINKED ISSUE #40310: Sliders won't move in short distances when competing with other drags
Recording: Screen Recording 2019-09-11 at 23 09 28 2019-09-11 23_12_13

## Steps to Reproduce

1. Get any Slider
2. Touch on the thumb
3. Move slightly to left or right (usually within the ripple area), it won't move.

I tried using the [sample code from Gallery app](https://github.com/flutter/flutter/blob/master/examples/flutter_gallery/lib/demo/material/slider_demo.dart) and it is fully reproducible. It seems, however, that the official Gallery app (which was last updated months ago) doesn't have this issue.

Edit: this is what cause the issue. kTouchSlop should be 0 for this scenario.
 https://github.com/flutter/flutter/blob/cd84ceaf5e3e8eee6a1e803938e176bdeb2efef0/packages/flutter/lib/src/gestures/monodrag.dart#L526

## Logs

```
➜ flutter doctor -v
[✓] Flutter (Channel dev, v1.10.1, on Mac OS X 10.14.6 18G95, locale en-BR)
    • Flutter version 1.10.1 at /usr/local/Cellar/flutter/dev
    • Framework revision ce45c2d3e6 (5 days ago), 2019-09-06 20:11:41 -0400
    • Engine revision b9ce2500d9
    • Dart version 2.5.0 (build 2.5.0-dev.4.0 be66176534)


[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    • Android SDK at /Users/bernardoferrari/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling
      support)
    • Platform android-29, build-tools 29.0.2
    • Java binary at: /Users/bernardoferrari/Library/Application
      Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/191.5791312/Android
      Studio 3.5 Preview.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_202-release-1483-b49-5587405)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 10.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 10.3, Build version 10G8
    • CocoaPods version 1.7.3

[✓] Android Studio (version 3.5)
    • Android Studio at /Users/bernardoferrari/Library/Application
      Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/191.5791312/Android
      Studio 3.5 Preview.app/Contents
    • Flutter plugin version 39.0.3
    • Dart plugin version 191.8423
    • Java version OpenJDK Runtime Environment (build
      1.8.0_202-release-1483-b49-5587405)

[✓] Android Studio
    • Android Studio at /Users/bernardoferrari/Library/Application
      Support/JetBrains/Toolbox/apps/AndroidStudio/ch-1/192.5863777/Android
      Studio 3.6 Preview.app/Contents
    • Flutter plugin version 39.0.4
    • Dart plugin version 192.6527
    • Java version OpenJDK Runtime Environment (build
      1.8.0_212-release-1586-b4-5784211)

[✓] IntelliJ IDEA Ultimate Edition (version 2019.2.2)
    • IntelliJ at /Users/bernardoferrari/Applications/JetBrains Toolbox/IntelliJ
      IDEA Ultimate.app
    • Flutter plugin version 38.2.4
    • Dart plugin version 192.6459

[✓] VS Code (version 1.38.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.4.1

[✓] Connected device (1 available)
    • SM G9600 • 23a1f4ee1a057ece • android-arm64 • Android 9 (API 28)

• No issues found!
```

### LINKED ISSUE #82153: Reduce the slop of dragged sliders
Currently the slop of already dragged Slider feels too big and slider feels sluggish, I propose that it should be reduced from default 18.0 to 8.0 or something like that

<details>
<summary>Code sample</summary>

```dart
import 'package:flutter/cupertino.dart';
import 'package:flutter/material.dart';

void main() => runApp(const MyApp());

class MyApp extends StatelessWidget {
  const MyApp({Key? key}) : super(key: key);

  static const String _title = 'Flutter Code Sample';

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: _title,
      home: Scaffold(
        appBar: AppBar(title: const Text(_title)),
        body: const MyStatefulWidget(),
      ),
    );
  }
}

class MyStatefulWidget extends StatefulWidget {
  const MyStatefulWidget({Key? key}) : super(key: key);

  @override
  _MyStatefulWidgetState createState() => _MyStatefulWidgetState();
}


class _MyStatefulWidgetState extends State<MyStatefulWidget> {
  double _currentSliderValue = 20;

  @override
  Widget build(BuildContext context) {
    return ListView(
      children: [
        Container(height: 100, color: Colors.red),
        Container(height: 100, color: Colors.green),
        Container(height: 100, color: Colors.blue),
        Container(height: 100, color: Colors.red),
        Container(height: 100, color: Colors.green),
        Container(height: 100, color: Colors.blue),
        GestureDetector(
          onHorizontalDragUpdate: (_) {},
          child: Slider(
            value: _currentSliderValue,
            min: 0,
            max: 100,
            onChangeStart: (_) {
              print('start');
            },
            onChangeEnd: (_) {
              print('end');
            },
            onChanged: (double value) {
              setState(() {
                _currentSliderValue = value;
              });
            },
          ),
        ),
        Container(height: 100, color: Colors.red),
        Container(height: 100, color: Colors.green),
        Container(height: 100, color: Colors.blue),
        Container(height: 100, color: Colors.red),
        Container(height: 100, color: Colors.green),
        Container(height: 100, color: Colors.blue),
      ],
    );
  }
}
```
</details>

https://user-images.githubusercontent.com/39104740/117580691-cef24680-b101-11eb-8c16-e51cfcb55c96.mp4

### PR DESCRIPTION
**Pull request by nt4f04uNd**
<strike>Fixes https://github.com/flutter/flutter/issues/82153</strike>
Fixes https://github.com/flutter/flutter/issues/40310

This is based of https://github.com/flutter/flutter/pull/82152 and https://github.com/flutter/flutter/pull/82131, so they should land first

Video before change is attached to the issue

Video after change

https://user-images.githubusercontent.com/39104740/117583708-6743f780-b111-11eb-880b-0c5a86e15906.mp4

## Pre-launch Checklist

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [ ] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [ ] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt.
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter/test/material/slider_test.dart [hunk 1]
```diff
@@ -908,6 +909,105 @@ void main() {
...
+    // Test when slider drag is not yet active
+
+    final TestGesture touchGestureNotActive = await tester.startGesture(center);
+
+    await touchGestureNotActive.moveBy(const Offset(kTouchSlop, 0.0));
+    expect(value, equals(0.0));
+
+    await touchGestureNotActive.moveBy(const Offset(1.0, 0.0));
+    expect(value, greaterThan(0.5));
+  });
```
Comment (nt4f04uNd) [inline packages/flutter/test/material/slider_test.dart:1009]: I noticed that other tests do `await gesture.up();` and sometimes `await tester.pumpAndSettle();` at their end. Is this required and why?

**Comment by nt4f04uNd** [comment]
Here a difference with active check and slop 6 and no active check and slop 6

| Check | No check |
| - | - |
| active_check | no_active_check |

File: packages/flutter/lib/src/material/slider.dart [hunk 2]
```diff
@@ -434,7 +434,12 @@ class Slider extends StatefulWidget {
   /// {@macro flutter.widgets.Focus.autofocus}
   final bool autofocus;
 
-  final _SliderType _sliderType ;
+  /// The distance pointer has to travel in global coordinate system after
+  /// drag was confirmed to be active, for framework to be confident that user wants to drag the slider.
+  @visibleForTesting
+  static const double activeDragSlop = 6.0;
```
Comment (goderbauer) [inline packages/flutter/lib/src/material/slider.dart:440]: Why 6?

**Review by goderbauer — COMMENTED** [review]
I don't think we want to make this change. Why does the drag gesture recognizer in the slider need to be special in its competition in the gesture arena over other drags? What "bug" is this fixing? It is working as intended that drags start with a delay if they are competing with other drags until we are sure the gesture is going in the right direction.

File: packages/flutter/lib/src/material/slider.dart (hunk 2 above)
Comment (nt4f04uNd) [reply packages/flutter/lib/src/material/slider.dart:440]: 18.0 causes https://github.com/flutter/flutter/issues/40310
But 0.0 would not allow vertical drags at all

So I picked 6.0 as a compromise value to remove the "jump", but still let the vertical drag

Comment (goderbauer) [reply packages/flutter/lib/src/material/slider.dart:440]: So, it still doesn't really fix https://github.com/flutter/flutter/issues/40310 then if you want to move even smaller distances.

**Comment by nt4f04uNd** [comment]
> Why does the drag gesture recognizer in the slider need to be special in its competition in the gesture arena over other drags?

When user holds his finger for the time sufficient for slider to become active, it's very likely that his intention to drag it, so we might want to reduce the slop to remove the "jump". In the GIFs above you can see that it still allows to scroll lists with no problem, as in not active state the slop is 18.0

**Pull request closed without merging**